lib/ansible/modules/net_tools/basics/slurp.py

```diff
@@ -77,14 +108,16 @@ def main():
 
     if not os.path.exists(source):
         module.fail_json(msg="file not found: %s" % source)
+    if os.path.isdir(source):
+        module.fail_json(msg="path is a directory: %s" % source)
     if not os.access(source, os.R_OK):
         module.fail_json(msg="file is not readable: %s" % source)
 
-    with open(source, 'rb') as source_fh:
-        source_content = source_fh.read()
-    data = base64.b64encode(source_content)
+    with io.BytesIO() as f:
+        checksum = b64encode_and_checksum(source, f)
+        data = to_native(f.getvalue()).strip()
```

Comment:
We will still end up with 2x here. I'm interested in investigating whether we can write this to a real file, and not memory, and then have our JSON encoder be able to stream from a file handle. I attempted this before, but need to look again.